be/src/vec/functions/array/function_array_distinct.h

```diff
@@ -0,0 +1,294 @@
+                         ColumnNullable* dest_nullable_col) {
+        const ColumnString* src_data_concrete = check_and_get_column<ColumnString>(&src_column);
+        if (!src_data_concrete) {
+            return false;
+        }
+
+        ColumnString& dest_column_string = reinterpret_cast<ColumnString&>(dest_column);
+        ColumnString::Offsets& column_string_offsets = dest_column_string.get_offsets();
+
+        const PaddedPODArray<UInt8>* src_null_map = nullptr;
+        if (src_nullable_col) {
+            src_null_map = &src_nullable_col->get_null_map_column().get_data();
+        }
+
+        PaddedPODArray<UInt8>* dest_null_map = nullptr;
+        if (dest_nullable_col) {
+            dest_null_map = &dest_nullable_col->get_null_map_column().get_data();
+        }
+
+        using Set =
+                HashSetWithSavedHashWithStackMemory<StringRef, StringRefHash, INITIAL_SIZE_DEGREE>;
+        Set set;
+
+        ColumnArray::Offset prev_src_offset = 0;
+        ColumnArray::Offset res_offset = 0;
+
+        for (auto curr_src_offset : src_offsets) {
+            set.clear();
+
+            size_t null_size = 0;
+            for (ColumnArray::Offset j = prev_src_offset; j < curr_src_offset; ++j) {
+                if (src_nullable_col && (*src_null_map)[j]) {
+                    if (dest_nullable_col && dest_null_map) {
+                        column_string_offsets.push_back(column_string_offsets.back());
+                        (*dest_null_map).push_back(true);
+                        null_size++;
+                    }
+                    continue;
+                }
+                StringRef src_str_ref = src_data_concrete->get_data_at(j);
+
+                if (!set.find(src_str_ref)) {
+                    set.insert(src_str_ref);
+                    dest_column_string.insert_data(src_str_ref.data, src_str_ref.size);
+                    if (dest_nullable_col && dest_null_map) {
+                        (*dest_null_map).push_back(false);
+                    }
+                }
+            }
+
+            res_offset += set.size() + null_size;
+            dest_offsets.push_back(res_offset);
+            prev_src_offset = curr_src_offset;
+        }
+        return true;
+    }
+
+    bool _execute_by_type(const IColumn& src_column, const ColumnArray::Offsets& src_offsets,
+                          IColumn& dest_column, ColumnArray::Offsets& dest_offsets,
+                          const ColumnNullable* src_nullable_col, ColumnNullable* dest_nullable_col,
+                          DataTypePtr& nested_type) {
+        bool res = false;
+        WhichDataType which(remove_nullable(nested_type)->get_type_id());
+        if (which.idx == TypeIndex::UInt8) {
+            res = _execute_number<UInt8>(src_column, src_offsets, dest_column, dest_offsets,
+                                         src_nullable_col, dest_nullable_col);
+        } else if (which.idx == TypeIndex::UInt16) {
```

Comment:
we do not support Unsigned data type, except UInt8 for Boolean.

Reply:
> we do not support Unsigned data type, except UInt8 for Boolean.

ok, remove it later.